metaflow/plugins/secrets/__init__.py

```diff
@@ -0,0 +1,14 @@
+import abc
+
+
+class SecretsProvider(abc.ABC):
```

Comment:
nit: we have not used this pattern in the code base. am fine with it just pointing out that it is not consistent.

Reply:
oh ya. Absolutely not with compatibility just with code style. I am fine either way. Just mentioning it. I try to stay within the "feel" of the code but there is an argument to be made that we can evolve it to make it better.